### ISSUE TITLE
JavaScript keywords are dropped

### ISSUE THREAD

**Issue by clecompt-msft** [digitalbazaar/jsonld.js#304]
Vocabulary keys matching JavaScript keywords are incorrectly dropped during expansion, compaction, etc. Consider the following:
```js
require('jsonld').expand({
  "@context": {
    "valueOf": "http://example.org/valueOf",
    "toString": "http://example.org/toString"
  },
  "valueOf": "first",
  "toString": "second"
}).then(result => console.log(JSON.stringify(result, null, 2)))
```
This returns the completely empty...
```json
[]
```
… instead of the expected...
```json
[
  {
    "http://example.org/toString": [
      {
        "@value": "second"
      }
    ],
    "http://example.org/valueOf": [
      {
        "@value": "first"
      }
    ]
  }
]
```

**Comment by davidlehn** [comment]
Yeah, that's a bug.  Somewhat horrifying that such a bug has slipped through for so long!  Thanks for pointing it out.  We'll release some test cases and a fix soon.

It's an issue of using a regular `Object` and `if(term in obj)` vs `if(obj.hasOwnProperty(term))`.  We had wanted to change code using checks like that to use `Map` which would avoid such issues.  But due to some active context implementation details that is a complex change so was put aside.  Will have to just selectively fix it for now.

**Comment by davidlehn** [comment]
Fix is in 1.6.0.  Thanks.
